src/rollback_to_stable/rts.c

```diff
@@ -105,6 +105,124 @@ __wt_rts_progress_msg(WT_SESSION_IMPL *session, WT_TIMER *rollback_start, uint64
     }
 }
 
+/*
+ * __wt_rts_thread_chk --
+ *     Check to decide if the RTS thread should continue running.
+ */
+bool
+__wt_rts_thread_chk(WT_SESSION_IMPL *session)
+{
+    return (F_ISSET(S2C(session), WT_CONN_RTS_THREAD_RUN));
+}
+
+/*
+ * __wt_rts_thread_run --
+ *     Entry function for an RTS thread. This is called repeatedly from the thread group code so it
+ *     does not need to loop itself.
+ */
+int
+__wt_rts_thread_run(WT_SESSION_IMPL *session, WT_THREAD *thread)
+{
+    WT_DECL_RET;
+    WT_RTS_WORK_UNIT *entry;
+
+    WT_UNUSED(thread);
+
+    /* Wait here. */
+    if (F_ISSET(S2C(session), WT_CONN_RTS_THREAD_RUN))
```

Comment:
Why we need to wait?

Reply:
If there are no files in the queue to process, it is better to wait here instead of occupying the CPU resources until a file is queued.

Reply:
What happens if WT_CONN_RTS_THREAD_RUN is not set? Should it be if the flag is not set, we wait until the flag is set?

Reply:
As this flag is set before the thread starts, it will wait. When this flag is turned off, the threads will stop.